lib/salus/scanners/report_go_dep.rb

```diff
@@ -1,44 +1,105 @@
 require 'toml'
 require 'salus/scanners/base'
+require 'salus/plugin_manager'
+require 'salus/report'
 
 # Report the use of Go packages captured in a Gopkg.lock files.
 # https://github.com/golang/dep
 
 module Salus::Scanners
   class ReportGoDep < Base
     def run
-      unless should_run?
+      if @repository.go_sum_present?
+        record_dep_from_go_sum
+      elsif @repository.dep_lock_present?
+        record_dep_from_go_lock_package
+      elsif @repository.go_mod_present?
+        record_dep_from_go_mod
+      else
         raise(
           InvalidScannerInvocationError,
-          'Cannot report on Go dependencies without a Gopkg.lock file.'
+          'Cannot report on Go dependencies without a Gopkg.lock, go.mod, or go.sum file'
         )
       end
+    end
+
+    def record_dep_from_go_sum
+      go_sum_path = "#{@repository.path_to_repo}/go.sum"
+      dep_list = []
+
+      File.foreach(go_sum_path).each("=\n") do |line|
+        line = line.strip
+        next if line.empty?
+
+        line_info = get_line_info(line)
+
+        dep_list.append(
+          {
+            "fullDependency" => line,
+            "name" => line_info[0],
```

Comment:
See note for a cleaner way to reference name / version vs the positional reference here